Modify runtime descriptors to support ADR 0004 (#3548)
oasisprotocol/oasis-core: docs/consensus/registry.md

A [runtime] is effectively a replicated application with shared state. The
registry resource describes a runtime's operational parameters, including its
identifier, kind, admission policy, committee scheduling, storage, governance
model, etc. For a full description of the runtime descriptor see
[the `Runtime` structure].

<!-- markdownlint-disable no-space-in-emphasis -->

> The chosen governance model indicates how the runtime descriptor can be updated
> in the future.
> There are currently three supported governance models:
>     * **entity governance** -- the runtime owner is the only one who can update
>       the runtime descriptor via `registry.RegisterRuntime` method calls.
>     * **runtime-defined governance** -- the runtime itself is the only one who
>       can update the runtime descriptor by emitting a runtime message.
>     * **consensus layer governance** -- only the consensus layer itself can
>       update the runtime descriptor either through a network upgrade or some
>       other future means.

Reviewer: ```suggestion
The chosen governance model indicates how the runtime descriptor can be updated
in the future.

There are currently three supported governance models:

* **Entity governance** where the runtime owner is the only one who can update
  the runtime descriptor via `registry.RegisterRuntime` method calls.

* **Runtime-defined governance** where the runtime itself is the only one who
  can update the runtime descriptor by emitting a runtime message.

* **Consensus layer governance** where only the consensus layer itself can
  update the runtime descriptor through network governance.
```

A [runtime] is effectively a replicated application with shared state. The
registry resource describes a runtime's operational parameters, including its
identifier, kind, admission policy, committee scheduling, storage, governance
model, etc. For a full description of the runtime descriptor see
[the `Runtime` structure].

> <!-- markdownlint-disable no-space-in-emphasis -->

Author: Yeah, it seems broken by default -- it complains about the spaces in emphasised phrases.
I've re-enabled it below now :)

> <!-- markdownlint-disable no-space-in-emphasis -->

Reviewer 2: thx!
